libs/causality/src/lib/CausalAnalysisDashboard/Controls/CausalAnalysisView/CausalIndividualView/LargeCausalIndividualView/largeCausalIndividualChartUtils.ts

```diff
@@ -0,0 +1,280 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { IComboBoxOption } from "@fluentui/react";
+import {
+  calculateBubblePlotDataFromErrorCohort,
+  Cohort,
+  FluentUIStyles,
+  hasAxisTypeUpdated,
+  ICausalAnalysisData,
+  ICausalAnalysisSingleData,
+  IDataset,
+  IGenericChartProps,
+  IHighchartBubbleSDKClusterData,
+  IHighchartsConfig,
+  IScatterPoint,
+  ISelectorConfig,
+  ITelemetryEvent,
+  JointDataset,
+  TelemetryEventName,
+  TelemetryLevels
+} from "@responsible-ai/core-ui";
+import { localization } from "@responsible-ai/localization";
+import _ from "lodash";
+
+import { CausalIndividualConstants } from "../CausalIndividualConstants";
+
+import {
+  ILargeCausalIndividualChartProps,
+  ILargeCausalIndividualChartState
+} from "./ILargeCausalIndividualChartSpec";
+
+export const absoluteIndexKey = "AbsoluteIndex";
+export const indexKey = "Index";
+
+export async function getBubblePlotData(
+  chartProps: IGenericChartProps,
+  cohort: Cohort,
+  jointDataset: JointDataset,
+  dataset: IDataset,
+  isLocalCausalDataLoading?: boolean,
+  requestBubblePlotData?:
+    | ((
+        filter: unknown[],
+        compositeFilter: unknown[],
+        xAxis: string,
+        yAxis: string,
+        abortSignal: AbortSignal
+      ) => Promise<IHighchartBubbleSDKClusterData>)
+    | undefined,
+  selectPointFromChartLargeData?: (data: IScatterPoint) => Promise<void>,
+  onBubbleClick?: (
+    scatterPlotData: IHighchartsConfig,
+    xSeries: number[],
+    ySeries: number[],
+    indexSeries: number[]
+  ) => void
+): Promise<IHighchartBubbleSDKClusterData | IHighchartsConfig | undefined> {
+  return await calculateBubblePlotDataFromErrorCohort(
+    cohort,
+    chartProps,
+    [],
+    jointDataset,
+    dataset,
+    isLocalCausalDataLoading,
+    true,
+    false,
+    requestBubblePlotData,
+    selectPointFromChartLargeData,
+    onBubbleClick,
+    undefined
+  );
+}
+
+export function shouldUpdateHighchart(
+  prevState: ILargeCausalIndividualChartState,
+  prevProps: ILargeCausalIndividualChartProps,
+  currentState: ILargeCausalIndividualChartState,
+  currentProps: ILargeCausalIndividualChartProps,
+  changedKeys: string[]
+): [boolean, boolean, boolean, boolean, boolean, boolean, boolean] {
+  const hasSelectedPointIndexesUpdated = !_.isEqual(
+    currentState.selectedPointsIndexes,
+    prevState.selectedPointsIndexes
+  );
+  const hasIsLocalExplanationsDataLoadingUpdated = !_.isEqual(
+    currentState.isLocalCausalDataLoading,
+    prevState.isLocalCausalDataLoading
+  );
+  const hasRevertToBubbleChartUpdated =
+    currentState.isRevertButtonClicked &&
+    prevState.isRevertButtonClicked !== currentState.isRevertButtonClicked;
```

Comment:
to be consistent, maybe update `!==` to  `!_.isEqual` for line 92 and 93?

Reply:
thanks Tong, updated